### PR TITLE
Created config class got `from_environ` method (fix #2)

### LINKED ISSUE #2: Can we avoid calling `environ.to_config`?
Seeing your presentation from PyCon 2018 (thanks a lot for that), I jumped nto your library to test `environ_config`. It works and helps.

Anyway, I got confused a bit and will try to describe that and propose alternative approach.

# What confused me

I refer to the example in README.

First I expected, that instance of `AppConfig` class will already be populated config object, but I found, that e.g. `env` property is a `Raise` instance.

Then I understood, `environ.to_config` shall be called. At the end, the directly instanciated `AppConfig` is not what one expects.

# Proposal 1: Full initialization using `os_environ`

What about:

```python
cfg = AppConfig()
```

or with explicit os_environ:

```python
os_environ={
 "APP_ENV": "dev",
 "APP_DB_HOST": "localhost",
 "LANG": "C",
 "APP_AWESOME": "yes",
 "SECRET_YOUR_APP_DB_PASSWORD": "s3kr3t",
}  
cfg = AppConfig(os_environ=os_environ)
```

pros:
- looks very simple and intuitive
cons:
- implementation may become complex (att values may come from two sources: `environ` and specific argument directly)

# Proposal 2: Add class method `from_environ`

It would work as follows:

```python
cfg = AppConfig.from_environ()
```

or using explicit `os_environ` value:

```python
os_environ={
 "APP_ENV": "dev",
 "APP_DB_HOST": "localhost",
 "LANG": "C",
 "APP_AWESOME": "yes",
 "SECRET_YOUR_APP_DB_PASSWORD": "s3kr3t",
}  
cfg = AppConfig.from_environ(os_environ)
```

One advantage would be, direct instantiation with explicit arguments could work as expected:

```python
cfg = AppConfig(env="dev", awesome=True, db=AppConfig.DB(host="localhost", password="s3kr3t"))

```
The `from_environ` could get `kwargs` argument to allow mixed creation:

```python
cfg = AppConfig.from_environ(os_environ=os_environ, kwargs=dict(env="dev", awesome=True))
```

pros:
- easy to create `AppConfig` instance without environmental variables
- implementation would be simpler
cons:
- direct instantiation could surprise (raising Exception on missing arguments).

Expected modifications would be:
- add class method (could be very similar to existing `environ.to_config`
- modify handling of arguments without defaults not to return `Raise` instance, but really raise an exception.

# Conclusions

First, I was in favour of Proposal 1, but the Proposal 2 seems simpler to implement and in a way looks cleaner. What do you think?

### PR DESCRIPTION
**Pull request by vlcinsky**
Modified `@environ.config` decorator to add classmethod `from_environ` to each created class.

This PR does not add any other method (such as `generate_help`)

### CODE REVIEW

File: tests/test_class_to_config.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import environ
+
+
+@environ.config(prefix="APP")
+class AppConfig(object):
+    host = environ.var("127.0.0.1")
+    port = environ.var(5000, converter=int)
+
+
+def test_default():
```
Comment (hynek) [inline tests/test_class_to_config.py]: Please add [test docstrings](https://jml.io/pages/test-docstrings.html) to all tests. Believe me, you'll be glad for it, if you come back to the code in a year.

File: README.rst [hunk 2]
```diff
@@ -37,8 +37,7 @@ environ_config: Configuration with env variables for Python.
   ...    lang = environ.var(name="LANG")  # It's possible to overwrite the names of variables.
   ...    db = environ.group(DB)
   ...    awesome = environ.bool_var()
-  >>> cfg = environ.to_config(
-  ...     AppConfig,
+  >>> cfg = AppConfig.from_environ(
```
Comment (hynek) [inline README.rst]: I don't think it's necessary to change the default here.

File: README.rst [hunk 3]
```diff
@@ -52,6 +51,23 @@ environ_config: Configuration with env variables for Python.
   >>> cfg.db.password
   's3kr3t'
 
+`AppConfig` instance can be also created by means of `environ.to_config` method:
```
Comment (hynek) [inline README.rst]: This is RST, so you'll have to use double backticks for code. :)

You don't have to add a whole example, it becomes hard to read. Just something like:

> `AppConfig.from_environ({...})` is equivalent to the code above, depending on your taste.

File: tests/test_class_to_config.py (hunk 1 above)
Comment (vlcinsky) [reply tests/test_class_to_config.py]: Done. This i like on PRs to other's projects - it pays back by giving me a lesson on good coding style. Thanks.

File: README.rst (hunk 2 above)
Comment (vlcinsky) [reply README.rst]: Agreed. Done.

File: README.rst (hunk 3 above)
Comment (vlcinsky) [reply README.rst]: Great use of [Occam's razor](https://en.wikipedia.org/wiki/Occam%27s_razor). Done.

File: tests/test_class_to_config.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    """Class based `from_environ` without `environ` argument.
```
Comment (hynek) [inline tests/test_class_to_config.py]: For the future, this is not my/our docstrig style (it looks squenched, doesn't it?) but I'll quickfix it myself – thanks!

**Pull request merged**